common/src/main/java/com/microsoft/identity/common/internal/telemetry/events/ApiEndEvent.java

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.telemetry.events;
+
+import android.support.annotation.NonNull;
+
+import com.microsoft.identity.common.exception.BaseException;
+import com.microsoft.identity.common.exception.UserCancelException;
+import com.microsoft.identity.common.internal.result.AcquireTokenResult;
+
+import static com.microsoft.identity.common.internal.telemetry.TelemetryEventStrings.*;
+
+public class ApiEndEvent extends BaseEvent {
+    private static final String TAG = ApiStartEvent.class.getSimpleName();
+
+    public ApiEndEvent() {
+        super();
+        names(TELEMETRY_EVENT_API_EVENT_END);
+    }
+
+    public ApiEndEvent putResult(@NonNull final AcquireTokenResult result) {
```

Comment:
Shouldn't this be `public ApiEndEvent putResult(@NonNull final ILocalAuthenticationResult result) {`

Beacuse if `result.getSucceeded()` we parse Auth and Token response in `ApiDispatcher` and create a `BaseException` . From there we can call `putException` method.

Otherwise here we just add ` put(TELEMETRY_KEY_IS_SUCCESSFUL, result.getSucceeded().toString());` and nothing else if `getLocalAuthenticationResult` is null

Reply:
I saw the return of the AT and ATS result is in type AcquireTokenResult

Reply:
yeah let's sort this out the after PR is in , primary feedback is acquiretoken result doesn't have the exception info to pass the telemetry